src/daemon/daemon.c

```diff
@@ -261,12 +261,22 @@ static gboolean do_exit(gpointer user_data)
 
 static void add_exit_timeout(NotifyDaemon* daemon)
 {
+	GdkDisplay *display = gdk_screen_get_display (gdk_screen_get_default());
+	/*Do not timeout if not running in X11
+	 *As wayland sessions are usually controlled
+	 *by the wayland compositor. We need to keep
+	 *the daemon running on wayland
+	 */
+	if (!(GDK_IS_X11_DISPLAY (display)))
+		return;
+
 	g_assert (daemon != NULL);
 
-	if (daemon->exit_timeout_source > 0)
+	if (daemon->exit_timeout_source == 0)
 		return;
 
-	daemon->exit_timeout_source = g_timeout_add_seconds(IDLE_SECONDS, do_exit, NULL);
```

Comment:
I use X11, it shouldn't be removed here

Reply:
That looks like an error left over from earlier experiments, I will fix that now

Reply:
Fixed that error, now we just have the return if not running under x11, should be no change in x11 behavior now